### PR TITLE
fix: accept Clever-compatible environment names

### PR DESCRIPTION
**Pull request by franky47**
## Summary

Accept Clever Cloud environment variable names containing dots and dashes so valid `setEnv` entries are no longer silently dropped. Invalid entries remain ignored, with their values redacted from warnings.

### CODE REVIEW

File: src/arguments.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ function throwMissingEnvVar(name: string): never {
   )
 }
 
-const ENV_LINE_REGEX = /^(\w+)=(.*)$/
+const ENV_LINE_REGEX = /^([a-zA-Z0-9-_.]+)=(.*)$/
```
Comment (franky47) [inline src/arguments.ts:32]: **Correctness:** Clever Tools 4.11's `validateName` accepts `__proto__`, so this regex correctly admits `__proto__=sentinel`; however, the reducer below starts from `{}` and `env[key] = value` invokes `Object.prototype.__proto__`'s setter rather than creating an own property. The parsed result has `Object.keys(extraEnv) === []`, and `Object.entries(extraEnv)` in `action.ts` therefore never sends this valid variable—silently reproducing the class of bug this change is meant to remove. I verified both `validateName('__proto__') === true` and the empty parsed result against this head. Please accumulate into a null-prototype object (for example `Object.create(null) as Record<string, string>`) or a `Map`, and add a regression test for `__proto__`.

**Review by franky47 — COMMENTED** [review]
## Review summary

Found one correctness gap in the otherwise exact Clever Tools 4.11 name-character alignment: the valid `__proto__` name is still silently discarded because the parser accumulates into a normal object. See the inline comment for the repro and suggested fix.

Verification: `CI=true pnpm test` passes (6 files, 128 tests, no type errors); I also compared this regex with `@clevercloud/client` 12.0.0's `validateName` contract used by Clever Tools 4.11.0 and exercised the edge case directly. Parser anchoring, malformed-key rejection, and warning value redaction otherwise look correct.

File: src/arguments.ts (hunk 1 above)
Comment (franky47) [reply src/arguments.ts:32]: Resolved in 667b183: `setEnv` now accumulates into a null-prototype object, preserving `__proto__` as an own enumerable variable. Added a regression test covering the parsed `Object.entries` result.

Comment (franky47) [reply src/arguments.ts:32]: Resolved in 292aa797105575a18c9b0f98d1e4dde831bb3e39 by explicitly rejecting the reserved `__proto__` name with a value-redacted warning. This avoids supporting a surprising prototype-sensitive key while preserving normal dotted and dashed Clever-compatible names.
